### PR TITLE
Websockets: Stop iteration when connection closes.

### PR DESCRIPTION
**Pull request by frederikaalund**
## What do these changes do?

Stops `async for msg in ws` iteration when `ws` is closed.

## Are there changes in behavior for the user?

The user no longer needs to handle `WSMsgType.CLOSED` events themselves (this was never documented anyhow).

## Related issue number

See issue #1144.

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

**Comment by asvetlov** [comment]
Please add a test.

**Comment by asvetlov** [comment]
`client_ws.py` should be updated also.

**Comment by frederikaalund** [comment]
> @frederikaalund did you give up the PR?

Sorry for the delay. It's been a busy month. I hope to have time to look at this on Monday.

**Comment by asvetlov** [comment]
@frederikaalund ping

**Comment by frederikaalund** [comment]
@asvetlov Sorry about the delay.
- I've updated `client_ws.py` as well.
- I've added two tests: One for the client interface and one for the server interface.

Due to the nature of the feature (which uses async for), said tests require Python 3.5 and above. Moreover, the tests will _not_ be able to run in Python 3.5 and below since the `async for` is not valid syntax in the previous versions. Of course, this only affects the development pipeline. I do not know of any workarounds to such syntax changes (I tried some `eval`/`exec` tricks but to no avail). I can't find a drop-in replacement for `async for` for Python 3.4 as can be done with `async def` and `async with`. If you know of one, I can update accordingly.

In summary, the test suite now requires Python 3.5 and above.

**Comment by asvetlov** [comment]
Please move your tests into `tests/test_py35` folder.
This directory is ignored when tests are executed by python 3.4

**Comment by frederikaalund** [comment]
> Please move your tests into tests/test_py35 folder.

Ah, so that is how it is done. I completely missed that.

**Comment by asvetlov** [comment]
LGTM.
The last thing: rules for PR was changed. New rules are here: https://github.com/KeepSafe/aiohttp/blob/master/PULL_REQUEST_TEMPLATE.md
Would you update `CHANGES.rst` and `CONTRIBUTORS.txt` as well?

**Comment by frederikaalund** [comment]
Everything should be in order now. Let me know if there is something else.

**Comment by asvetlov** [comment]
Awesome!
Thank you very much!
